File: js_validation/pardox/src/_bindings/databases.js
'use strict';

module.exports = function bindDatabases(_lib) {
    // MySQL
    _lib.pardox_read_mysql        = _lib.raw.func('void * pardox_read_mysql(const char * conn_str, const char * query)');
    _lib._raw_pardox_write_mysql  = _lib.raw.func('int64_t pardox_write_mysql(void * mgr, const char * conn_str, const char * table, const char * mode, const char * conflict_cols_json)');
    _lib._raw_pardox_execute_mysql = _lib.raw.func('int64_t pardox_execute_mysql(const char * conn_str, const char * query)');

    // SQL Server
    _lib.pardox_read_sqlserver        = _lib.raw.func('void * pardox_read_sqlserver(const char * conn_str, const char * query)');
    _lib._raw_pardox_write_sqlserver  = _lib.raw.func('int64_t pardox_write_sqlserver(void * mgr, const char * conn_str, const char * table, const char * mode, const char * conflict_cols_json)');
    _lib._raw_pardox_execute_sqlserver = _lib.raw.func('int64_t pardox_execute_sqlserver(const char * conn_str, const char * query)');

    // MongoDB
    _lib.pardox_read_mongodb        = _lib.raw.func('void * pardox_read_mongodb(const char * conn_str, const char * db_dot_collection)');
    _lib._raw_pardox_write_mongodb  = _lib.raw.func('int64_t pardox_write_mongodb(void * mgr, const char * conn_str, const char * db_dot_collection, const char * mode)');
    _lib._raw_pardox_execute_mongodb = _lib.raw.func('int64_t pardox_execute_mongodb(const char * conn_str, const char * database, const char * command_json)');
};

File: py_validation/pardox/_bindings/databases.py
def bind(lib, c_void_p, c_char_p, c_int32, c_double, c_longlong, c_size_t, c_int64, c_int16):
    if hasattr(lib, 'pardox_read_mysql'):
        lib.pardox_read_mysql.argtypes = [c_char_p, c_char_p]
        lib.pardox_read_mysql.restype = c_void_p

    if hasattr(lib, 'pardox_write_mysql'):
        lib.pardox_write_mysql.argtypes = [c_void_p, c_char_p, c_char_p, c_char_p, c_char_p]
        lib.pardox_write_mysql.restype = c_longlong

    if hasattr(lib, 'pardox_execute_mysql'):
        lib.pardox_execute_mysql.argtypes = [c_char_p, c_char_p]
        lib.pardox_execute_mysql.restype = c_longlong

    if hasattr(lib, 'pardox_read_sqlserver'):
        lib.pardox_read_sqlserver.argtypes = [c_char_p, c_char_p]
        lib.pardox_read_sqlserver.restype = c_void_p

    if hasattr(lib, 'pardox_write_sqlserver'):
        lib.pardox_write_sqlserver.argtypes = [c_void_p, c_char_p, c_char_p, c_char_p, c_char_p]
        lib.pardox_write_sqlserver.restype = c_longlong

    if hasattr(lib, 'pardox_execute_sqlserver'):
        lib.pardox_execute_sqlserver.argtypes = [c_char_p, c_char_p]
        lib.pardox_execute_sqlserver.restype = c_longlong

    if hasattr(lib, 'pardox_read_mongodb'):
        lib.pardox_read_mongodb.argtypes = [c_char_p, c_char_p]
        lib.pardox_read_mongodb.restype = c_void_p

    if hasattr(lib, 'pardox_write_mongodb'):
        lib.pardox_write_mongodb.argtypes = [c_void_p, c_char_p, c_char_p, c_char_p]
        lib.pardox_write_mongodb.restype = c_longlong

    if hasattr(lib, 'pardox_execute_mongodb'):
        lib.pardox_execute_mongodb.argtypes = [c_char_p, c_char_p, c_char_p]
        lib.pardox_execute_mongodb.restype = c_longlong
